src/Microsoft.PowerShell.Commands.Utility/commands/utility/Get-Error.cs

```diff
@@ -0,0 +1,90 @@
+{
+    /// <summary>
+    /// Class for Get-Error implementation.
+    /// </summary>
+    [Experimental("Microsoft.PowerShell.Utility.PSGetError", ExperimentAction.Show)]
+    [Cmdlet(VerbsCommon.Get, "Error", HelpUri = "https://docs.microsoft.com/powershell/module/microsoft.powershell.utility/get-error?view=powershell-7&WT.mc_id=ps-gethelp", DefaultParameterSetName = NewestParameterSetName)]
+    public sealed class GetErrorCommand : PSCmdlet
+    {
+        internal const string ErrorParameterSetName = "Error";
+        internal const string NewestParameterSetName = "Newest";
+        internal const string AliasNewest = "Last";
+
+        /// <summary>
+        /// Gets or sets the error object to resolve.
+        /// </summary>
+        [Parameter(Position = 0, ValueFromPipeline = true, ParameterSetName = ErrorParameterSetName)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of error objects to resolve starting with newest first.
+        /// </summary>
+        [Parameter(ParameterSetName = NewestParameterSetName)]
+        [Alias(AliasNewest)]
+        [ValidateRange(1, int.MaxValue)]
+        public int Newest { get; set; } = 1;
+
+        /// <summary>
+        /// Process the error object.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            var errorRecords = new List<object>();
+            var index = 0;
+
+            if (InputObject != null)
+            {
+                if (InputObject.BaseObject is Exception || InputObject.BaseObject is ErrorRecord)
+                {
+                    errorRecords.Add(InputObject);
+                }
+            }
+            else
+            {
+                var errorVariable = SessionState.PSVariable.Get("error");
+                var count = Newest;
+                ArrayList errors = (ArrayList)errorVariable.Value;
+                if (count > errors.Count)
+                {
+                    count = errors.Count;
+                }
+
+                while (count > 0)
+                {
+                    errorRecords.Add(errors[index]);
+                    index++;
+                    count--;
+                }
+            }
+
+            index = 0;
+            bool addErrorIdentifier = errorRecords.Count > 1 ? true : false;
+
+            foreach (object errorRecord in errorRecords)
+            {
+                PSObject obj = PSObject.AsPSObject(errorRecord);
+                obj.TypeNames.Insert(0, "System.Management.Automation.ErrorRecord#GetError");
+
+                // Remove some types so they don't get rendered by those formats
+                obj.TypeNames.Remove("System.Management.Automation.ErrorRecord");
+                obj.TypeNames.Remove("System.Exception");
+
+                if (addErrorIdentifier)
+                {
+                    obj.Properties.Add(new PSNoteProperty("PSErrorIdentifier", index++));
```

Comment:
`PSErrorIdentifier` sounds like it contains an actual identifier for an error record. Identifiers are usually unique, such that you can re-retrieve data using that identifier. In this case, the identifier is always changing, because it's really an index. How about `PSErrorIndex` instead, since that's what it is?

Reply:
Also, this could be handled more generically by deriving an `ErrorArrayList` class from `ArrayList`, and overriding `this[int index]` in that class to return the annotated `PSObject` with the `PSErrorIndex` property added to it. That way you allow people used to using `$error` to continue doing so, while also allowing users to use `Get-Error`, and both get back the same data.

Reply:
Changes to `$error` should be a separate PR from this.  In this case, the index for formatting is really used as a separator for grouping if you request to view multiple errors at once.  As you note, since the index changes with any new error, it's actually not that useful.